tzatziki-spring-kafka/src/main/java/com/decathlon/tzatziki/steps/KafkaSteps.java

```diff
@@ -177,6 +177,12 @@ public void we_publish_on_a_topic(Guard guard, String name, String topic, Object
         guard.in(objects, () -> publish(name, topic, content, null));
     }
 
+    @SneakyThrows
+    @When(THAT + GUARD + A + RECORD + " (?:is|are)? published on the " + QUOTED_CONTENT + " topic:$")
```

Comment:
Maybe instead of declaring a new steps for template variables, we can add a new constant TEMPLATE_OR_VARIABLE_PATTERN and replace it in the we_publish_on_a_topic steps to have a generic steps that can handle VARIABLE AND TEMPLATE VARIABLE.

Reply:
good idea 👍🏼 I change that